### ISSUE TITLE
GameObjectDisplayCondition.db2 Support

### ISSUE THREAD

**Issue by Selenium** [TrinityCore/TrinityCore#30483]
### Description

Since TWW the Holidays GameObject in Dornogal have a dynamic DisplayId and a single spawn, nothing handle that.
This system also handle the end chests in Delves. 

For the exemple i will use the GameObject 454734 named `HDressing 2.0 - Standing, Banner01 - Trial of Style/Noblegarde/Children/Pilgrim/Midsummer/Brewfest - JZB`
the name itself show one gameobject have multiple purpose

### How it's working db2 wise:
https://wago.tools/db2/GameObjectDisplayCondition?build=11.0.2.56311&filter[GameObjectID]=454734&page=1&sort[Field_10_2_7_54171_002]=desc

### Few field here can be named; 
Field_10_2_7_54171_001: Containt the DisplayId of the gameobject (we can confirm it by sniffing each event on retail)
Field_10_2_7_54171_002: is a WorldStateId (credit to @Nyr97 ) (see under for values)
Field_10_2_7_54171_003: Unknow purpose, only a field have something else than 0 and it look like a dev test
Field_10_2_7_54171_004: Float value may related to a size override

#### How ? 

Knowing the field Field_10_2_7_54171_001 is a DisplayId we can search it in GameObjectDisplayInfo.db2
https://wago.tools/db2/GameObjectDisplayInfo?build=11.0.2.56311&filter[ID]=91502&page=1
this DB2 link the DisplayId to a FileDataId
https://wago.tools/files?build=11.0.2.56311&search=5282591

`world/expansion09/doodads/festival/10xp_childrensweek_bannerstanding01.m2`

By doing that on all DisplayId matching the WorldStateId we can deduce the names based on the file naming

```
1368 = xmas
2016 = easter
2174 = childrensweek
3555 = summerfest
4546 = beerfest
6572 = thanksgiving
7523 = valentinesday
22303 = masquerade

not holiday related
42057/42058 = earthen
43337 = nerubian
43338 = kobyss
43339 = rootlands
43340 = arathor
43341 = kobold
43366 = nerubian
```

From Firestorm-servers team, with love

### Expected behaviour

.

### Steps to reproduce the problem

.

### Branch

master

### TC rev. hash/commit

[508396c](https://github.com/TrinityCore/TrinityCore/commit/508396c26e0f6c2f8fcb7d3f35f4970932623d6a)

### Operating system

None

### Custom changes

None

**Comment by Nyr97** [comment]
Precisely, Field_10_2_7_54171_002 is a WorldStateExpression, which translates to WorldStates that only go true or false. We might wait until Christmas is up and you'll notice that the WorldState 841 is 1 then.

However, you can easily check this by going into a delve, let's say, a Nerubian one, and the expression 43337 or 43366 will be set to true in those only.

**Comment by Selenium** [comment]
update with WSE result for holidays
